File: app/src/main/java/com/shevart/androidcorelearn/fragments/mobile_and_tablet_ui/DetailScreenActivity.java
package com.shevart.androidcorelearn.fragments.mobile_and_tablet_ui;

import android.os.Bundle;

import com.shevart.androidcorelearn.R;
import com.shevart.androidcorelearn.common.AbsActivity;
import com.shevart.androidcorelearn.common.SimpleItem;

public class DetailScreenActivity extends AbsActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_detail_screen);
        enableToolbarBackButton();

        final DetailScreenFragment fragment = (DetailScreenFragment) getSupportFragmentManager()
                .findFragmentById(R.id.fragmentDetail);
        fragment.update(getSimpleItemFromIntent());
    }

    private SimpleItem getSimpleItemFromIntent() {
        return getIntent().getParcelableExtra(SimpleItem.class.getSimpleName());
    }
}

File: app/src/main/java/com/shevart/androidcorelearn/fragments/mobile_and_tablet_ui/DetailScreenActivity.kt
package com.shevart.androidcorelearn.fragments.mobile_and_tablet_ui

import android.os.Bundle

import com.shevart.androidcorelearn.R
import com.shevart.androidcorelearn.common.AbsActivity
import com.shevart.androidcorelearn.common.SimpleItem

class DetailScreenActivity : AbsActivity() {

    private val simpleItemFromIntent: SimpleItem
        get() = intent.getParcelableExtra(SimpleItem::class.java.simpleName)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_detail_screen)
        enableToolbarBackButton()

        val fragment = supportFragmentManager
                .findFragmentById(R.id.fragmentDetail) as DetailScreenFragment
        fragment.update(simpleItemFromIntent)
    }
}